test/e2e/cruise/multiplevirtualnodes/multiple_virtualnodes_test.go

```diff
@@ -0,0 +1,264 @@
+		if err != nil {
+			return err
+		}
+		if len(vNodes.Items) != expectedNum {
+			return fmt.Errorf("Found %d VirtualNodes after cleanup, expected %d", len(vNodes.Items), expectedNum)
+		}
+
+		nodes, err := getters.ListLiqoNodes(ctx, cl)
+		if err != nil {
+			return err
+		}
+		if len(nodes.Items) != expectedNum {
+			return fmt.Errorf("Found %d Liqo nodes after cleanup, expected %d", len(nodes.Items), expectedNum)
+		}
+
+		return nil
+	}
+)
+
+var _ = Describe("Liqo E2E", func() {
+	Context("Multiple VirtualNodes and ResourceSlices", func() {
+		AfterEach(func() {
+			for i := range testContext.Clusters {
+				deleteTestResourceSlices(testContext.Clusters[i].ControllerClient)
+			}
+
+			for i := range testContext.Clusters {
+				Eventually(func() error {
+					return checkCleanUp(testContext.Clusters[i].ControllerClient,
+						testContext.Clusters[i].Role, testContext.Clusters[i].NumPeeredProviders)
+				}, timeout, interval).Should(Succeed())
+			}
+		})
+
+		When("Peerings are established", func() {
+			It("Should have the right number of resourceslices", func() {
+				for i := range testContext.Clusters {
+					cluster := testContext.Clusters[i]
+					switch cluster.Role {
+					case liqov1alpha1.ConsumerRole:
+						resSlices, err := getters.ListResourceSlicesByLabel(ctx, cluster.ControllerClient,
+							corev1.NamespaceAll, liqolabels.LocalLabelSelector())
+						Expect(err).To(Not(HaveOccurred()))
+						Expect(len(resSlices)).To(Equal(cluster.NumPeeredProviders))
+					case liqov1alpha1.ProviderRole:
+						resSlices, err := getters.ListResourceSlicesByLabel(ctx, cluster.ControllerClient,
+							corev1.NamespaceAll, liqolabels.RemoteLabelSelector())
+						Expect(err).To(Not(HaveOccurred()))
+						Expect(len(resSlices)).To(Equal(cluster.NumPeeredConsumers))
+					default:
+						resSlices, err := getters.ListResourceSlicesByLabel(ctx, cluster.ControllerClient, corev1.NamespaceAll, labels.Everything())
+						Expect(err).To(Not(HaveOccurred()))
+						Expect(len(resSlices)).To(BeZero())
+					}
+				}
+			})
+		})
+
+		When("Adding multiple ResourceSlices for each providers on multiple providers", func() {
+
+			It("Should replicate the ResourceSlices to the provider, and create VirtualNodes and Nodes", func() {
+				// On every consumer cluster, create a ResourceSlice for each provider cluster.
+				for i := range testContext.Clusters {
+					if testContext.Clusters[i].Role == liqov1alpha1.ConsumerRole {
+						consumer := testContext.Clusters[i]
+						for j := range testContext.Clusters {
+							if testContext.Clusters[j].Role == liqov1alpha1.ProviderRole {
+								provider := testContext.Clusters[j]
+								createTestResourceSlice(consumer.ControllerClient, fmt.Sprintf("rs-test-%s", provider.Cluster), provider.Cluster, true)
+							}
+						}
+					}
+				}
+
+				// Test that every ResourceSlices, VirtualNodes and Nodes are created.
+				Eventually(func() error {
```

Comment:
Split this eventually in multiple ones, in order to better understand from the test logs which one is failed